### PR TITLE
[CI] fix LinuxCondaEnv builds (install dpcpp-cpp-rt with testing requirements)

### PR DESCRIPTION
**Pull request by icfaust**
## Description

The dpcpp-cpp-rt is needed to get the code functional for testing, but is not required for building. Install it at the proper step, which will fix the missing library issue (libsvml.so)

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

File: .ci/pipeline/build-and-test-lnx.yml [hunk 1]
```diff
@@ -55,6 +55,7 @@ steps:
       . /usr/share/miniconda/etc/profile.d/conda.sh
       conda activate CB
       bash .ci/scripts/setup_sklearn.sh $(SKLEARN_VERSION)
+      conda install -q -y -c conda-forge dpcpp-cpp-rt=2025.0
```
Comment (david-cortes-intel) [inline .ci/pipeline/build-and-test-lnx.yml:58]: Why does it need to be versioned?

Comment (icfaust) [reply .ci/pipeline/build-and-test-lnx.yml:58]: Very good question, it was previously always pinned (and thought it was best practice). Thoughts @Alexsandruss ? I have no problem either way.

Comment (Alexsandruss) [reply .ci/pipeline/build-and-test-lnx.yml:58]: dpcpp releases might be unevenly rolled out in different distribution channels, version pinning is required to avoid random failures related to incompatibilities between different versions in that case. Maybe, it is possible to reduce dpcpp pin to major version only.

Comment (david-cortes-intel) [reply .ci/pipeline/build-and-test-lnx.yml:58]: But this har 'conda-forge' hard-coded into it.

Comment (icfaust) [reply .ci/pipeline/build-and-test-lnx.yml:58]: I think he doesn't mean conda exclusively by distribution channel, but also include all the possibilities found here: https://www.intel.com/content/www/us/en/developer/tools/oneapi/dpc-compiler-download.html

Comment (napetrov) [reply .ci/pipeline/build-and-test-lnx.yml:58]: in general we would like reproducibility of results in main pipeline.  with all changes of tooling done through commits. this is something that ideally should be tracked by automation, so might be putting this to environment.yml and using from there

**Comment by icfaust** [comment]
#2440 is a much better approach
